android/src/com/mapswithme/maps/intent/Factory.java

```diff
@@ -475,10 +538,16 @@ protected MapTask createTargetTask(@NonNull String url)
                        .authority("").build();
 
       String query = coreUri.getLastPathSegment();
-      MapTask statisticTask = StatisticMapTaskWrapper.wrap(new OpenUrlTask(coreUri.toString(),
-                                                                             TextUtils.isEmpty(query)
-                                                                             ? Statistics.ParamValue.UNKNOWN :
-                                                                             query));
+      MapTask statisticTask = StatisticMapTaskWrapper.wrap(new OpenUrlTask(coreUri.toString())
```

Comment:
Иннер класс в Java держит неявно ссылку на внешний класс, т.к. на OldCoreLinkAdapterProcessor. Т.е. при сериализации экземпляра OpenUrlTask java пытается сериализовать и экземпляр внешнего класса. 

Serialization of inner classes (i.e., nested classes that are not static member classes), including local and anonymous classes, is strongly discouraged for several reasons. Because inner classes declared in non-static contexts contain implicit non-transient references to enclosing class instances, serializing such an inner class instance will result in serialization of its это отсюда https://docs.oracle.com/javase/8/docs/platform/serialization/spec/serial-arch.html#a4539

Иннер классы опасная вещь в Java и лучше их использовать только при явной необходимости. В нашем случае рекомендую тебе передавать как и было через конструктор названия параметра, если это возможно.

Reply:
Вернул проброс параметра

Reply:
Спасибо

Reply:
Только я понял это немного иначе - OpenUrlTask статический класс, поэтому с ним все нормально, но если перегрузить его метод на месте, то создается анонимный класс, который не является статическим и он держит ссылку. Жалко, что анонимные классы нельзя делать статическими.